src/plugins/data/public/data_sources/datasource_selector/types.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiComboBoxOptionOption, EuiComboBoxSingleSelectionShape } from '@elastic/eui';
+import { GenericDataSource } from '../datasource_services';
+
+export interface DataSourceGroup {
+  label: string;
+  options: DataSourceOption[];
+}
+
+export interface DataSourceOption {
```

Comment:
The convention in OSD is to call these "Options".
```suggestion
export interface DataSourceOptions {
```

Reply:
do you have a reference? I believe in oui it's `Option`, like 
```typescript
	export interface EuiComboBoxOptionOption<T = string | number | string[] | undefined> extends CommonProps, Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'value'> {
	    isGroupLabelOption?: boolean;
	    label: string;
	    key?: string;
	    options?: Array<EuiComboBoxOptionOption<T>>;
	    value?: T;
	}
```
and i think this type is somehow related to `EuiComboBoxOptionOption`

Reply:
OUI is not OSD’s role model. AFK ATM, but search for Options { and you will find hundreds

Reply:
i see, make sense. i think the code need better clarity on if it will be passed as a combo box option